pkg/tsdb/kairosdb/kairosdb.go

```diff
@@ -0,0 +1,186 @@
+
+	result := &tsdb.BatchResult{}
+
+	kairosdbQuery := NewQueryBuilder()
+	kairosdbQuery.SetAbsoluteStart(queryContext.TimeRange.MustGetFrom().UTC())
+	kairosdbQuery.SetAbsoluteEnd(queryContext.TimeRange.MustGetTo().UTC())
+	plog.Debug("Start_absolute", kairosdbQuery.AbsoluteStart)
+	plog.Debug("end_absolute", kairosdbQuery.AbsoluteEnd)
+
+	for _, query := range queries {
+		metric := e.buildMetric(query)
+		kairosdbQuery.AddMetric(metric)
+	}
+
+	if setting.Env == setting.DEV {
+		plog.Debug("KairosDb request", "params", kairosdbQuery)
+	}
+
+	req, err := e.createRequest(kairosdbQuery)
+	if err != nil {
+		result.Error = err
+		return result
+	}
+
+	res, err := ctxhttp.Do(ctx, e.httpClient, req)
+	if err != nil {
+		result.Error = err
+		return result
+	}
+
+	queryResult, err := e.parseResponse(res)
+	if err != nil {
+		return result.WithError(err)
+	}
+
+	result.QueryResults = queryResult
+	return result
+}
+
+func (e *KairosdbExecutor) createRequest(kairosdbQuery QueryBuilder) (*http.Request, error) {
+	u, _ := url.Parse(e.Url)
+	u.Path = path.Join(u.Path, "api/v1/datapoints/query/")
+
+	postData, err := kairosdbQuery.Build()
+	plog.Info("kairosdbQuery  - ", "build", string(postData))
+
+	req, err := http.NewRequest(http.MethodPost, u.String(), strings.NewReader(string(postData)))
+	if err != nil {
+		plog.Info("Failed to create request", "error", err)
+		return nil, fmt.Errorf("Failed to create request. error: %v", err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	if e.BasicAuth {
+		req.SetBasicAuth(e.BasicAuthUser, e.BasicAuthPassword)
+	}
+
+	return req, err
+}
+
+func (e *KairosdbExecutor) parseResponse(res *http.Response) (map[string]*tsdb.QueryResult, error) {
+
+	queryResults := make(map[string]*tsdb.QueryResult)
+	queryRes := tsdb.NewQueryResult()
+
+	body, err := ioutil.ReadAll(res.Body)
+	defer res.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode/100 != 2 {
+		plog.Info("Request failed", "status", res.Status, "body", string(body))
+		return nil, fmt.Errorf("Request failed status: %v", res.Status)
+	}
+
+	queryResponse := response.NewQueryResponse(res.StatusCode)
+	plog.Info("http response before Unmarshalling ", "marshal", string(body))
+	err = json.Unmarshal(body, queryResponse)
+
+	if err != nil {
+		plog.Info("Failed to unmarshal opentsdb response", "error", err, "status", res.Status, "body", string(body))
+		return nil, err
+	}
+
+	plog.Info("http response after unmarshalling", "unmarshall", queryResponse)
+
+	for _, query := range queryResponse.QueriesArr {
+		for _, result := range query.ResultsArr {
+			series := tsdb.TimeSeries{
+				Name: result.Name,
+			}
+			for _, datapoint := range result.DataPoints {
+				series.Points = append(series.Points, tsdb.NewTimePoint(null.FloatFrom(datapoint.Float64Value()), datapoint.Timestamp()))
+			}
+
+			queryRes.Series = append(queryRes.Series, &series)
+		}
+	}
+	queryResults["A"] = queryRes
```

Comment:
@ddhirajkumar : That is how it is done in all other executors. AFAIK it is because there is only one query allowed for alert execution. You can opentsdb.go, influxdb.go, prometheus.go....they all do the same way.